templates/bot/ts/default/config.ts

```diff
@@ -0,0 +1,13 @@
+const config = {
+  botId: process.env.BOT_ID,
+  botPassword: process.env.BOT_PASSWORD,
+  clientId: process.env.M365_CLIENT_ID,
+  clientSecret: process.env.M365_CLIENT_SECRET,
+  tenantId: process.env.M365_TENANT_ID,
+  authorityHost: process.env.M365_AUTHORITY_HOST,
+  initiateLoginEndpoint: process.env.INITIATE_LOGIN_ENDPOINT,
+  apiEndpoint: process.env.API_ENDPOINT,
```

Comment:
Where is the config `apiEndpoint` used?

Reply:
It would be used if developer wants to call Azure Functions in Bot, so list all possible environment variables here, and it is aligned with the `.env.teamsfx.local` generated by local-debug

Reply:
Do we need to list all the settings or we focus on what is used by the template?

Reply:
For `.env.teamsfx.local`, I think we need to list all the settings, and align with local-debug.
For `config.ts`, it is worthy to discuss. Add @jidddddd for PM's thought on this.
cc @blackchoey

Reply:
If the code doesn't use apiEndpoint, I would prefer to remove it in config.ts to avoid confusing.

For .env.teamsfx.local, my opinion is the keys in it should align with `config.ts`, which means it should also not have `apiEndpoint`. Otherwise user maybe confused where does the code consumes `apiEndpoint`.

How do you think not having `apiEndpoint` in scaffolded `.env.teamsfx.local` but an empty `apiEndpoint` is added to `.env.teamsfx.local` during local debug?

Also cc @swatDong to see what's the effort to not generate apiEndpoint if the source code does not need it. Though the priority is low from my perspective.

Reply:
In that case, for the alignment, we need to remove other keys which are not used as well, then only keep `BOT_ID` and `BOT_PASSWORD` in `.env.teamsfx.local` and  `config.ts` file?

Reply:
In my opinion, yes. We just need to keep bot id & password.

Reply:
I think it's better to only list variables as needed if possible. If the user wants to add extra features, they would follow a how-to guide on what and where to add variables.

cc: @MuyangAmigo, @swatDong, @blackchoey

Reply:
And another thing is, can we merge the template change now? Is there any dependency needs ready first?

Reply:
@blackchoey @jidddddd Updated.

Reply:
> And another thing is, can we merge the template change now? Is there any dependency needs ready first?

@adashen De you mean merge this PR into `dev` branch? I think there is no dependency.